src/cpp/client.cpp

```diff
@@ -932,4 +884,78 @@ void Client::_unpack_dataset_metadata(DataSet& dataset,
                                       reply[i+1].str_len());
     }
     return;
+}
+
+TensorBase* Client::_get_tensorbase_obj(const std::string& name)
+{
+    std::string g_key = this->_build_tensor_key(name, true);
+
+    CommandReply reply = this->_redis_server->get_tensor(g_key);
+
+    std::vector<size_t> dims =
+        CommandReplyParser::get_tensor_dims(reply);
+
+    TensorType type =
+         CommandReplyParser::get_tensor_data_type(reply);
+
+    std::string_view blob =
+        CommandReplyParser::get_tensor_data_blob(reply);
+
+    if(dims.size()<=0)
+        throw std::runtime_error("The number of dimensions of the "\
+                                "fetched tensor are invalid: " +
+                                std::to_string(dims.size()));
+
+    for(size_t i=0; i<dims.size(); i++) {
+        if(dims[i]<=0) {
+        throw std::runtime_error("Dimension " +
+                                 std::to_string(dims[i]) +
```

Comment:
Shouldn't this be just `i` instead of `dims[i]`?

Reply:
Yes, you are correct.  The error output should be ``i``.  Thanks!